dll/win32/shell32/brfolder.cpp

```diff
@@ -253,6 +253,73 @@ BrFolder_GetName(
     return bSuccess;
 }
 
+static BOOL
+BrFolder_IsTreeItemInEnum(
+    _Inout_ BrFolder *info,
+    _In_ HTREEITEM hItem,
+    _Inout_ IEnumIDList *pEnum)
+{
+    BrItemData *pItemData = BrFolder_GetItemData(info, hItem);
+    if (!pItemData)
+        return FALSE;
+
+    pEnum->Reset();
+
+    BOOL ret = FALSE;
+    CComHeapPtr<ITEMIDLIST_RELATIVE> pidlTemp;
+    while (S_OK == pEnum->Next(1, &pidlTemp, NULL))
+    {
+        if (ILIsEqual(pidlTemp, pItemData->pidlChild))
+        {
+            ret = TRUE;
+            break;
+        }
+        pidlTemp.Free();
+    }
+
+    return ret;
+}
+
+static BOOL
+BrFolder_TreeItemHasThisChild(BrFolder *info, HTREEITEM hItem, PITEMID_CHILD pidlChild)
+{
+    HTREEITEM hNextItem;
+    for (hItem = TreeView_GetChild(info->hwndTreeView, hItem); hItem; hItem = hNextItem)
+    {
+        hNextItem = TreeView_GetNextSibling(info->hwndTreeView, hItem);
+
```

Comment:
or
```suggestion
    for (hItem = TreeView_GetChild(info->hwndTreeView, hItem); hItem;
         hItem = TreeView_GetNextSibling(info->hwndTreeView, hItem))
    {
```
unless one of the function calls in the for-loop is modifying the treeview list?

Reply:
Fixed.